### ISSUE TITLE
Add support for syscall expressions with more than two arguments

### ISSUE THREAD

**Issue by ash-nolan** [ash-nolan/nova#4]
The `syscall` expression currently supports a maximum of two arguments (syscall number + one syscall parameter), but should allow up to seven syscall arguments (syscall number + six syscall parameters) on x86-64 Linux[1].

References:
[1]: [syscall(2)](https://manpages.debian.org/stable/manpages-dev/syscall.2.en.html)
[2]: [syscalls(2)](https://manpages.debian.org/stable/manpages-dev/syscalls.2.en.html)
[3]: [Anatomy of a system call, part 1](https://lwn.net/Articles/604287/)
[4]: [Anatomy of a system call, part 2](https://lwn.net/Articles/604515/)
[5]: [Linux System Call Table for x86-64](http://blog.rchapman.org/posts/Linux_System_Call_Table_for_x86_64/)

**Comment by ash-nolan** [comment]
A good candidate to test with all six parameters might be `sys_mmap`:
```c
void *mmap(void *addr, size_t length, int prot, int flags, int fd, off_t offset);
```
%rax | System call | %rdi | %rsi | %rdx | %r10 | %r8 | %r9
-- | -- | -- | -- | -- | -- | -- | --
9 | sys_mmap | unsigned long addr | unsigned long length | unsigned long prot | unsigned long flags | unsigned long fd | unsigned long offset

Which could be written as:
```nova
func mmap(addr: usize, length: usize, prot: ssize, flags: ssize, fd: ssize, offset: usize) ssize {
    return syscall(9s, addr, length, prot, flags, fd, offset);
}
```

**Comment by ash-nolan** [comment]
Closed by 7f63ce19838823ba687a4504a70cf62b52a5b189.
